lua/orgmode/config/init.lua

```diff
@@ -406,7 +415,7 @@ end
 ---@param amount number
 ---@return string
 function Config:get_indent(amount)
-  if self.opts.org_indent_mode == 'indent' then
+  if vim.b.org_indent_mode then
```

Comment:
Are you sure we will always have access to the buffer variable?

I would prefer if we stick only with configuration checks for now, and we can later add the buffer level toggling.

Reply:
It makes it much easier to use the buffer variable later on when we add toggling functionality.

I figured I'd bite the bullet now and take care of it as it's pretty fundamental to how it works.

If you'd prefer for me to undo that and have a separate PR with those checks I can do that. I have tests locally that have been rewritten now that support the buffer variable and they're all green currently.

Reply:
The latest commits are making heavy use of the `vim.b.org_indent_mode` variable, it's not that hard to substitute it for something else if you'd prefer me to not use a buffer variable for now. I can definitely use a pure config variable or something similar.